src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -462,18 +463,37 @@ export class ResultsViewPageStore {
 
     readonly clinicalDataForSamples = remoteData<ClinicalData[]>({
         await: () => [
+            this.studies,
             this.samples
         ],
         invoke: () => {
-            const filter: ClinicalDataMultiStudyFilter = {
-                attributeIds: ["CANCER_TYPE", "CANCER_TYPE_DETAILED"],
-                identifiers: this.samples.result.map((s: Sample) => ({entityId: s.sampleId, studyId: s.studyId}))
-            };
-            return client.fetchClinicalDataUsingPOST({
-                clinicalDataType: "SAMPLE",
-                clinicalDataMultiStudyFilter: filter,
-                projection: "DETAILED"
-            });
+            // single study query endpoint is optimal so we should use it
+            // when there's only one study
+            const clinicalDataTypes: string[] = ["CANCER_TYPE", "CANCER_TYPE_DETAILED"];
+
+            if (this.studies.result.length === 1) {
+                const study = this.studies.result[0];
+                const filter: ClinicalDataSingleStudyFilter = {
+                    attributeIds: clinicalDataTypes,
+                    ids: this.samples.result.map((s: Sample) =>s.sampleId)
+                };
+                return client.fetchAllClinicalDataInStudyUsingPOST({
+                    studyId:study.studyId,
+                    projection:'DETAILED',
```

Comment:
I don't think we need DETAILED projection for this call

Reply:
ok